lib/osf-components/addon/components/file-browser/file-rename-modal/component.ts

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable eqeqeq */
+/* eslint-disable no-console */
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+
+import RouterService from '@ember/routing/router-service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+
+import buildChangeset from 'ember-osf-web/utils/build-changeset';
+import { validatePresence } from 'ember-changeset-validations/validators';
+import { ValidationObject } from 'ember-changeset-validations';
+import OsfStorageManager from 'osf-components/components/storage-provider-manager/osf-storage-manager/component';
+import Toast from 'ember-toastr/services/toast';
+import Store from '@ember-data/store';
+import OsfAdapter from 'ember-osf-web/adapters/osf-adapter';
+import CurrentUser from 'ember-osf-web/services/current-user';
+import { restartableTask } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import OsfStorageFile from 'ember-osf-web/packages/files/osf-storage-file';
+import Intl from 'ember-intl/services/intl';
+
+interface Args {
+    manager: OsfStorageManager;
+    item: OsfStorageFile;
```

Comment:
Since Brian's PR for generalizing storage provider is in, you may do this
```suggestion
    manager: StorageManager;
    item: StorageFile;
```

Reply:
Ok, that had broken horribly when I tried it previously. Let me pull and merge changes and test this again.